src/common/badblock_poolset.c

```diff
@@ -126,3 +127,63 @@ os_badblocks_check_poolset(struct pool_set *set, int create)
 
 	return (cfcb.n_files_bbs > 0);
 }
+
+/*
+ * os_badblocks_clear_poolset_cb -- (internal) callback clearing bad blocks
+ *                                  in the given file
+ */
+static int
+os_badblocks_clear_poolset_cb(struct part_file *pf, void *arg)
+{
+	LOG(3, "part_file %p arg %p", pf, arg);
+
+	int *create = arg;
+
+	if (pf->is_remote) { /* XXX not supported yet */
+		ERR(
+			"clearing bad blocks in remote replicas is not supported yet -- '%s'",
+			pf->path);
+		errno = ENOTSUP;
+		return 0;
```

Comment:
if we set an errno, shouldn't we return a -1? How would the user know to check the errno value otherwise?